[regression] Missing -lm

Build of the version ```0.3.5``` fails on FreeBSD without setting ```-lm``` externally:
```
[ 98%] Linking C executable openblas_utest
cd /usr/ports/math/openblas/work/.build/utest && /usr/local/bin/cmake -E cmake_link_script CMakeFiles/openblas_utest.dir/link.txt --verbose=1
/usr/bin/cc -O2 -pipe -fno-omit-frame-pointer  -fstack-protector -fno-strict-aliasing  -DEXPRECISION -fPIC -DNO_AVX -DNO_AVX2 -DSMP_SERVER -DMAX_CPU_NUMBER=8 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.5\"" -O2 -pipe -fno-omit-frame-pointer  -fstack-protector -fno-strict-aliasing   -Wl,-rpath=/usr/local/lib/gcc8  -L/usr/local/lib/gcc8 -B/usr/local/bin -fstack-protector CMakeFiles/openblas_utest.dir/utest_main.c.o CMakeFiles/openblas_utest.dir/test_amax.c.o CMakeFiles/openblas_utest.dir/test_rotmg.c.o CMakeFiles/openblas_utest.dir/test_rot.c.o CMakeFiles/openblas_utest.dir/test_axpy.c.o CMakeFiles/openblas_utest.dir/test_dsdot.c.o CMakeFiles/openblas_utest.dir/test_swap.c.o CMakeFiles/openblas_utest.dir/test_dotu.c.o CMakeFiles/openblas_utest.dir/test_potrs.c.o  -o openblas_utest -Wl,-rpath,/usr/ports/math/openblas/work/.build/lib ../lib/libopenblas.so.0.3 -pthread 
/usr/local/bin/ld: CMakeFiles/openblas_utest.dir/test_potrs.c.o: undefined reference to symbol 'cabs@@FBSD_1.0'
/usr/local/bin/ld: /lib/libm.so.5: error adding symbols: DSO missing from command line
cc: error: linker command failed with exit code 1 (use -v to see invocation)
*** [utest/openblas_utest] Error code 1
```

You use the function ```cabs``` which requires ```-lm```. It should be set in cmake files.

--- a/cmake/os.cmake
+++ b/cmake/os.cmake
@@ -8,6 +8,11 @@ if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
   set(NO_EXPRECISION 1)
 endif ()
 
+if (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD|OpenBSD|NetBSD|DragonFly")
+  set(EXTRALIB "${EXTRALIB} -lm")
+  set(NO_EXPRECISION 1)
+endif ()
+
 if (${CMAKE_SYSTEM_NAME} STREQUAL "AIX")
   set(EXTRALIB "${EXTRALIB} -lm")
 endif ()

